nsqd/http.go

```diff
@@ -596,3 +533,80 @@ func (s *httpServer) printStats(stats []TopicStats, health string, startTime tim
 	}
 	return buf.Bytes()
 }
+
+func (s *httpServer) doConfig(w http.ResponseWriter, req *http.Request, ps httprouter.Params) (interface{}, error) {
+	opt := ps.ByName("opt")
+
+	if req.Method == "PUT" {
+		// add 1 so that it's greater than our max when we test for it
+		// (LimitReader returns a "fake" EOF)
+		readMax := s.ctx.nsqd.getOpts().MaxMsgSize + 1
+		body, err := ioutil.ReadAll(io.LimitReader(req.Body, readMax))
+		if err != nil {
+			return nil, http_api.Err{500, "INTERNAL_ERROR"}
+		}
+		if int64(len(body)) == readMax || len(body) == 0 {
+			return nil, http_api.Err{413, "INVALID_VALUE"}
+		}
+
+		var v interface{}
+		err = json.Unmarshal(body, &v)
+		if err != nil {
+			return nil, http_api.Err{400, "INVALID_VALUE"}
+		}
+
+		opts := *s.ctx.nsqd.getOpts()
+		switch opt {
+		case "nsqlookupd_tcp_addresses":
+			var addrs []string
```

Comment:
I feel like this would be cleaner to Unmarshal  directly to the expected type inside this switch (ie: all the type checks go away)

```go
var addrs []string
if err := json.Unmarshal(body, &addrs); err != nil {
    return nil, http_api.Err{400, "INVALID_VALUE"}
}
opts.NSQLookupdTCPAddresses = addrs
```

any reason not to?